blog/_posts/2025-04-22-Enhance-security-OS-Security-Patch.md

```diff
@@ -0,0 +1,67 @@
+date: 2025-04-22
+authors:
+- Kaarthikeyan Subramanian
+- Ye Wang
+categories: 
+- Security
+- Operations
+---
+
+
+## Traditional patching and the need for Managed patching.
+
+Operating System (OS) security patches are critical for safeguarding systems against vulnerabilities that malicious actors could exploit. These patches help ensure your system remains protected against emerging threats. Traditionally, customers have relied on nightly updates, such as [unattended upgrades in Ubuntu](https://help.ubuntu.com/community/AutomaticSecurityUpdates) or [Automatic Guest OS Patching](https://learn.microsoft.com/en-us/azure/virtual-machines/automatic-vm-guest-patching) at the virtual machine (VM) level. However, when kernel security packages were updated, a host machine reboot was often required, typically managed using tools like [kured](https://github.com/kubereboot/kured). 
+
+This approach, while effective, introduced challenges. Untested or uncontrolled security packages occasionally caused outages, emphasizing the need for a more reliable and managed patching mechanism at the node level. Additionally, maintaining reboot daemonsets like [kured](https://github.com/kubereboot/kured) added operational overhead for many customers. Ideally, customers prefer managed Kubernetes services, such as Azure Kubernetes Service (AKS), to handle OS security patching comprehensively and seamlessly.
+
+## Automatic Node OS Security Patching mechanisms at AKS
+
+AKS provides two managed and tested mechanisms to deliver the latest security packages to your Node Operating System. 
+
+**Automatic Node Image Channel** - AKS updates nodes weekly with a newly patched VHD for security and bug fixes. This update follows maintenance windows and upgrade configuration settings. Automatic Node image upgrades are supported as long as the cluster's Kubernetes minor version is in support. These AKS-tested node images are fully managed and applied with safe deployment practices.
+
+**OS Security Patch Channel** - Several customers may need only the security packages for their OS without additional bug fixes and updates. The OS Security Patch channel provides a fully managed, attended Node OS security-only solution. The Security-Patch channel reimages nodes only when necessary and provides live security patching updates with zero disruption, respecting planned maintenance windows and follows azure safe deployment practices.
+
+
+## When to use OS Security Patch or Automatic Node Image Channel?
+
+Choosing between the OS Security Patch channel and the Automatic Node Image channel depends on your specific requirements and operational constraints. Here's a breakdown based on common scenarios:
+
+- **Speed of Patching is Paramount**: If addressing critical vulnerabilities (CVEs) promptly is a priority, the OS Security Patch channel is the better choice. It delivers updates 1 to 2 weeks faster than the Node Image channel, ensuring your workloads remain secure against emerging threats.
```

Comment:
Again here I suggest you position this as a clean comparison, vs claiming one is faster than the other.

As such, suggest:

- replacing promptly with as fast as possible. (we can't assume what a user considers prompt, they might consider 2wks a lot faster than what they need)

- Security patching delivers security fixes in XX days, node image delivers it in 1-2weeks

Or so